### PR TITLE
Secure config api

### PR DESCRIPTION
**Pull request by KirilKabakchiev**

### CODE REVIEW

File: pkg/authn/basic.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ func basicAuthnMatchers() []web.FilterMatcher {
 	return []web.FilterMatcher{
 		{
 			Matchers: []web.Matcher{
-				web.Path(web.OSBURL + "/**"),
```
Comment (pankrator) [inline pkg/authn/basic.go:64]: Not sure this should be changed

**Review by pankrator — DISMISSED** [review]

File: pkg/authn/basic.go (hunk 1 above)
Comment (KirilKabakchiev) [reply pkg/authn/basic.go:64]: agent doesnt have osb api anymore right ?

File: pkg/sbproxy/sbproxy.go [hunk 2]
```diff
@@ -106,14 +110,29 @@ func New(ctx context.Context, cancel context.CancelFunc, settings *Settings, pla
 	log.AddHook(&logging.ErrorLocationHook{})
 
 	util.HandleInterrupts(ctx, cancel)
+	filters := []web.Filter{
+		&filters.Logging{},
+	}
+	authnSettings := settings.Authentication
+	if len(authnSettings.User) != 0 && len(authnSettings.Password) != 0 {
```
Comment (NickyMateev) [inline pkg/sbproxy/sbproxy.go:117]: General question: Shouldn't we always expect to have authentication settings provided?

And then just introduce a Validate method on the new authentication settings to use before constructing the basic and oidc filters?

Comment (KirilKabakchiev) [reply pkg/sbproxy/sbproxy.go:117]: This way we will make it mandatory for whoever uses the OSS agent to have to configure authentication even if he doesnt really want to use the config api

**Review by NickyMateev — DISMISSED** [review]

**Review by NickyMateev — APPROVED** [review]

**Pull request merged**